docs/specs/devcontainerjson-reference.md

```diff
@@ -0,0 +1,141 @@
+| `mounts` | array | An array of additional mount points to add to the container when created. Each value is a string that accepts the same values as the [Docker CLI `--mount` flag](https://docs.docker.com/engine/reference/commandline/run/#add-bind-mounts-or-volumes-using-the---mount-flag). Environment and [pre-defined variables](#variables-in-devcontainerjson) may be referenced in the value. For example:<br />`"mounts": ["source=${localWorkspaceFolder}/app-scripts,target=/usr/local/share/app-scripts,type=bind,consistency=cached"]` |
+| `workspaceMount` | string | Requires `workspaceFolder` be set as well. Overrides the default local mount point for the workspace when the container is created. Supports the same values as the [Docker CLI `--mount` flag](https://docs.docker.com/engine/reference/commandline/run/#add-bind-mounts-or-volumes-using-the---mount-flag). Environment and [pre-defined variables](#variables-in-devcontainerjson) may be referenced in the value. For example: <br />`"workspaceMount": "source=${localWorkspaceFolder}/sub-folder,target=/workspace,type=bind,consistency=cached", "workspaceFolder": "/workspace"` |
+| `workspaceFolder` | string | Requires `workspaceMount` be set. Sets the default path that `devcontainer.json` supporting services / tools should open when connecting to the container. Defaults to the automatic source code mount location. |
+| `runArgs` | array | An array of [Docker CLI arguments](https://docs.docker.com/engine/reference/commandline/run/) that should be used when running the container. Defaults to `[]`. For example, this allows ptrace based debuggers like C++ to work in the container:<br /> `"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ]` . |
+
+### Docker Compose specific properties
+
+| Property | Type | Description |
+|----------|------|-------------|
+| `dockerComposeFile` | string,<br>array | **Required** when [using Docker Compose](https://code.visualstudio.com/docs/remote/create-dev-container#_use-a-dockerfile). Path or an ordered list of paths to Docker Compose files relative to the `devcontainer.json` file. Using an array is useful [when extending your Docker Compose configuration](https://code.visualstudio.com/docs/remote/create-dev-container#_extend-your-docker-compose-file-for-development). The order of the array matters since the contents of later files can override values set in previous ones.<br>The default `.env` file is picked up from the root of the project, but you can use `env_file` in your Docker Compose file to specify an alternate location.<br>Note that the array syntax will execute the command without a shell. You can [learn more](#formatting-string-vs-array-properties) about formatting string vs array properties. |
```

Comment:
@chuxel do you think we should remove this link to the VS Code docs?

> [when extending your Docker Compose configuration](https://code.visualstudio.com/docs/remote/create-dev-container#_extend-your-docker-compose-file-for-development)

It's a somewhat long section, so had a couple thoughts:
- Option 1: Could copy content into this doc
- Since above would make this already lengthy doc longer, could instead:
     - Option 2: Create a separate doc we could link here, with this link's content and potentially other material, like https://code.visualstudio.com/docs/remote/create-dev-container#_use-a-dockerfile from below
     - Option 3: Leave link as-is
     - Option 4: Remove link and disregard the additional content it links to (if it's more VS Code based / we don't think it's essential)

Thoughts? I thought option 2 seemed reasonable to avoid lengthening this doc, still linking relevant content, and being extensible to add more content over time.

Reply:
Yeah that's another good one.  From this spot we may just be able start out by referencing https://docs.docker.com/compose/extends/#multiple-compose-files

The VS Code content provides a more specific example but links to this same spot. I think the question is whether we end up moving some of the "advanced" content and things like this here or not.  We could certainly do a "hello world" using the dev container CLI.

Reply:
Updated the link to https://docs.docker.com/compose/extends/#multiple-compose-files. 

> I think the question is whether we end up moving some of the "advanced" content and things like this here or not.

What you say makes sense - added a note to our ongoing plan about this so that we can remember it when we are ready.

> I do see value in some basic "creating your first dev container" content with links to product specific content.

Do you envision this as separate "creating your first dev container" guides for different products/editors? Perhaps could create a centralized page that links out to specific info about each tool?

Reply:
What I suspect we'll want to do here is have generic getting started info with the CLI, then links to product specific getting started. e.g. "In this tutorial, we'll use the dev container CLI to create a development container. You can learn more about getting started with other tools..." which would go to their respective docs.  Just thinking logistically that tools need to be able to update their getting started steps on a schedule independant from our merging of web content. So this is the index to that kind of contetnt... but there's a "neutral" example using the reference implementation.

Reply:
Makes sense! For instance, link to the Remote - Containers ["Create a development container"](https://code.visualstudio.com/docs/remote/create-dev-container) guide from the Remote - Containers section of supporting-tools.md.

Thinking it makes sense to start a "Getting Started with the dev container CLI" once the reference implementation is published - @chrmarti @Chuxel let me know if we should start any of that page now.

Reply:
I think we're probably okay to add that later like you mention.  Agree @chrmarti?

Reply:
I agree! (Late answer, catching up with notifications.)